resources/deployment_resource.go

```diff
@@ -50,6 +56,14 @@ func (d Deployment) MarshalJSON() ([]byte, error) {
 		ccDeployment.Strategy = d.Strategy
 	}
 
+	var b DeploymentOpts
+	if d.Options != b {
+		ccDeployment.Options = &d.Options
+		if d.Options.MaxInFlight < 1 {
+			ccDeployment.Options.MaxInFlight = 1
```

Comment:
[concern, blocking] what happens when we talk to an older CAPI, I.e. one that doesn't have max-in-flight? in #3088 we approached this by assuming that a value of 0 was "max-in-flight unset in the response" but I'm reading this block to mean that we'll always be setting max-in-flight to the default value of 1 and including it in a POST to CAPI
edit: Unless I am missing something and this object is only for unmarshalling the CAPI response

Reply:
The concern is valid. When max-in-flight gets to this point with a value we just send it to CAPI and CAPI will return an error, which I am going to address in the next commit.
Also did change the way we are defining if --max-in-flight is present or not by using a pointer instead of the -1 value which should make everything much easier to reason about.

Reply:
agreed on the nullable pointer being easy to reason about. If we handle the version issue in followup work I'm happy to consider this resolved.